### PR TITLE
Defect/de2543 sign in to ac

### PR DESCRIPTION
**Pull request by dtkocher**
***Rally***: https://rally1.rallydev.com/#/27593764268d/detail/defect/82800886212
***Build***: http://mp-ci.centralus.cloudapp.azure.com/viewLog.html?buildId=13471&tab=buildResultsDiv&buildTypeId=Code_Development_SignInCheckIn_Defect

### CODE REVIEW

File: SignInCheckIn/MinistryPlatform.Translation/Repositories/EventRepository.cs [hunk 1]
```diff
@@ -97,6 +97,11 @@ public MpEventDto CreateSubEvent(string token, MpEventDto mpEventDto)
             return _ministryPlatformRestRepository.UsingAuthenticationToken(token).Create(mpEventDto, _eventColumns);
         }
 
+        public MpEventDto UpdateSubEvent(string token, MpEventDto mpEventDto)
```
Comment (jpc11623) [inline SignInCheckIn/MinistryPlatform.Translation/Repositories/EventRepository.cs]: Do we have an update event call already? Can this just be called update event?

File: SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs [hunk 2]
```diff
@@ -803,10 +803,10 @@ public void CheckAcEventDuringCurrentService()
             // Arrange
             _applicationConfiguration.Setup(m => m.AdventureClubEventTypeId).Returns(20);
 
-            DateTime currentMpEventDtoStartTime = new DateTime(2016, 12, 1, 9, 0, 0);
-            DateTime futureMpEventDtoStartTime = new DateTime(2016, 12, 1, 11, 0, 0);
+            var currentMpEventDtoStartTime = new DateTime(2016, 12, 1, 9, 0, 0);
```
Comment (jpc11623) [inline SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs]: I'm concerned about using var for all the variable definitions in unit tests - I noticed an issue a couple of sprints ago where the Mock or NUnit List type was being using in place of System...List, which caused problems when I added a type comparison to the test.

File: SignInCheckIn/MinistryPlatform.Translation/Repositories/EventRepository.cs (hunk 1 above)
Comment (dtkocher) [inline SignInCheckIn/MinistryPlatform.Translation/Repositories/EventRepository.cs]: We do not currently have an update event call.

File: SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs [hunk 3]
```diff
@@ -883,41 +883,41 @@ public void CheckAcEventAfterCurrentService()
...
 
-            EventDto signingInEventDto = new EventDto
+            var signingInEventDto = new EventDto
             {
                 EventId = 1234567,
                 EventSiteId = 8
             };
 
-            ParticipantEventMapDto participantEventMapDto = new ParticipantEventMapDto
+            var participantEventMapDto = new ParticipantEventMapDto
```
Comment (samueljmello) [inline SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs]: Is there a reason to remove all the type-safe references?

File: SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs (hunk 2 above)
Comment (dtkocher) [inline SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs]: In this particular case at least var vs none var did not cause any issues.  I think we should investigate the issue if the test is failing due to var as we use that throughout the code and if it is causing issues in tests we should be concerned with the code outside of tests having the same problem.  In the case you mentioned my guess is if we had specified `new System.List<T>` var would not have caused and issue.

File: SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs (hunk 3 above)
Comment (dtkocher) [inline SignInCheckIn/SignInCheckIn.Tests/Services/ChildSigninServiceTest.cs]: Linting in c# was complaining.

**Pull request merged**